File: migrate_handouts.mjs
import fs from 'fs/promises';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const handoutsDir = path.join(__dirname, 'handouts');
const markdownDir = path.join(__dirname, 'te-kete-ako-astro', 'src', 'content', 'handouts');

async function createMarkdownFromHtml(htmlFilePath, markdownDirPath) {
  try {
    const htmlContent = await fs.readFile(htmlFilePath, 'utf-8');

    const titleMatch = htmlContent.match(/<title>(.*?)<\/title>/);
    const title = titleMatch ? titleMatch[1] : 'Untitled';

    const bodyMatch = htmlContent.match(/<body>([\s\S]*)<\/body>/);
    const bodyContent = bodyMatch ? bodyMatch[1].trim() : '';

    const fileName = title.toLowerCase().replace(/[^a-z0-9]+/g, '-') + '.md';
    const markdownFilePath = path.join(markdownDirPath, fileName);

    const frontmatter = `---
title: "${title}"
description: "A handout on ${title}."
tags: []
level: 1
learningArea: "Uncategorized"
---

${bodyContent}
`;

    await fs.writeFile(markdownFilePath, frontmatter, 'utf-8');
    console.log(`Migrated ${path.basename(htmlFilePath)}`);
  } catch (error) {
    console.error(`Error migrating ${htmlFilePath}:`, error);
  }
}

async function main() {
  try {
    await fs.mkdir(markdownDir, { recursive: true });
    const files = await fs.readdir(handoutsDir);

    for (const file of files) {
      if (file.endsWith('.html')) {
        const htmlFilePath = path.join(handoutsDir, file);
        await createMarkdownFromHtml(htmlFilePath, markdownDir);
      }
    }
  } catch (error) {
    console.error('Error during migration:', error);
  }
}

main();

File: migrate_handouts.py
import os
from bs4 import BeautifulSoup
import re

def create_markdown_from_html(html_file_path, markdown_dir):
    with open(html_file_path, 'r', encoding='utf-8') as f:
        html_content = f.read()

    soup = BeautifulSoup(html_content, 'html.parser')

    title = soup.title.string if soup.title else 'Untitled'
    body_content = soup.body.prettify() if soup.body else ''

    # Basic cleaning of the title for the filename
    file_name = re.sub(r'[^a-z0-9]+', '-', title.lower()) + '.md'
    markdown_file_path = os.path.join(markdown_dir, file_name)

    # Create frontmatter
    frontmatter = f"""---
title: "{title}"
description: "A handout on {title}."
tags: []
level: 1
learningArea: "Uncategorized"
---

{body_content}
"""

    with open(markdown_file_path, 'w', encoding='utf-8') as f:
        f.write(frontmatter)

def main():
    script_dir = os.path.dirname(os.path.abspath(__file__))
    handouts_dir = os.path.join(script_dir, 'handouts')
    markdown_dir = os.path.join(script_dir, 'te-kete-ako-astro', 'src', 'content', 'handouts')

    if not os.path.exists(markdown_dir):
        os.makedirs(markdown_dir)

    for filename in os.listdir(handouts_dir):
        if filename.endswith('.html'):
            html_file_path = os.path.join(handouts_dir, filename)
            create_markdown_from_html(html_file_path, markdown_dir)
            print(f"Migrated {filename}")

if __name__ == '__main__':
    main()
